clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -1057,6 +1059,205 @@ public void testDescribeConfigsUnrequested() throws Exception {
+            assertEquals(emptySet(), descriptionsMap.get("/var/data/kafka").replicaInfos().keySet());
+
+            Map<Integer, Map<String, LogDirDescription>> allDescriptions = result.allDescriptions().get();
+            assertEquals(Collections.singleton(0), allDescriptions.keySet());
+            Map<String, LogDirDescription> allMap = allDescriptions.get(0);
+            assertNotNull(allMap);
+            assertEquals(Collections.singleton("/var/data/kafka"), allMap.keySet());
+            assertEquals(KafkaStorageException.class, allMap.get("/var/data/kafka").error().getClass());
+            assertEquals(emptySet(), allMap.get("/var/data/kafka").replicaInfos().keySet());
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testDescribeLogDirsOfflineDirDeprecated() throws ExecutionException, InterruptedException {
+        List<Integer> brokers = singletonList(0);
+
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareResponseFrom(new DescribeLogDirsResponse(
+                    new DescribeLogDirsResponseData().setResults(asList(new DescribeLogDirsResponseData.DescribeLogDirsResult()
+                            .setErrorCode(Errors.KAFKA_STORAGE_ERROR.code())
+                            .setLogDir("/var/data/kafka")
+                            .setTopics(emptyList())
+                    ))), env.cluster().nodeById(0));
+            DescribeLogDirsResult result = env.adminClient().describeLogDirs(brokers);
+
+            Map<Integer, KafkaFuture<Map<String, DescribeLogDirsResponse.LogDirInfo>>> deprecatedValues = result.values();
+            assertEquals(Collections.singleton(0), deprecatedValues.keySet());
+            assertNotNull(deprecatedValues.get(0));
+            Map<String, DescribeLogDirsResponse.LogDirInfo> valuesMap = deprecatedValues.get(0).get();
+            assertEquals(Collections.singleton("/var/data/kafka"), valuesMap.keySet());
+            assertEquals(Errors.KAFKA_STORAGE_ERROR, valuesMap.get("/var/data/kafka").error);
+            assertEquals(emptySet(), valuesMap.get("/var/data/kafka").replicaInfos.keySet());
+
+            Map<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>> deprecatedAll = result.all().get();
+            assertEquals(Collections.singleton(0), deprecatedAll.keySet());
+            Map<String, DescribeLogDirsResponse.LogDirInfo> allMap = deprecatedAll.get(0);
+            assertNotNull(allMap);
+            assertEquals(Collections.singleton("/var/data/kafka"), allMap.keySet());
+            assertEquals(Errors.KAFKA_STORAGE_ERROR, allMap.get("/var/data/kafka").error);
+            assertEquals(emptySet(), allMap.get("/var/data/kafka").replicaInfos.keySet());
+        }
+    }
+
+    @Test
+    public void testDescribeReplicaLogDirs() throws ExecutionException, InterruptedException {
```

Comment:
This is not due to your PR but shall we add a unit test which uses multiple brokers?

Reply:
I added it to the existing test. Due to the new helper methods I felt this didn't really complicate the test very much and is also allows us to cover the case where the RPC returns `STORAGE_ERROR`.